### PR TITLE
Fix outstanding errors for Bevy 0.13

### PR DESCRIPTION
**Pull request by Ant59**
I've made an attempt to fix up the remaining compilation, linting and testing errors in Lightyear with the Bevy 0.13 update.

However, I think for the Leafwing Input Manager feature to compile, that crate needs to update.

### CODE REVIEW

File: lightyear/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ webtransport = [
   "dep:tokio",
   "dep:ring",
 ]
-leafwing = ["dep:leafwing-input-manager", "lightyear_macros/leafwing"]
+leafwing = ["dep:leafwing-input-manager", "lightyear_macros/leafwing", "bevy/bevy_asset"]
```
Comment (cBournhonesque) [inline lightyear/Cargo.toml:33]: is bevy asset necessary for leafwing?

Comment (Ant59) [reply lightyear/Cargo.toml:33]: It was necessary for the tests to run in the Leafwing feature (for use bevy::asset::AsyncReadExt), but I think it's not necessary for the build. I'm not entirely sure how to convince Cargo to only include a dependency for test with a certain feature though.

**Comment by cBournhonesque** [comment]
Thanks! I integrated most of your changes.
I don't wnat to change bitcode too much as that part of the code can be a bit fragile

**Comment by Ant59** [comment]
No problem. The bitcode changes were just ported from the latest commit on the bitcode repo to fix issues in Rust nightly. Works fine with Rust stable until next version.

**Pull request closed without merging**